### ISSUE TITLE
No results using eq filter on edge

### ISSUE THREAD

**Issue by boedy** [dgraph-io/dgraph#5475]
Initial discussion with @anurags92 in [Slack channel ](https://dgraph.slack.com/archives/C13LH03RR/p1589897029050300)

### What version of Dgraph are you using?
docker.io/dgraph/dgraph:v1.2.2

### Have you tried reproducing the issue with the latest release?
No.

### What is the hardware spec (RAM, OS)?
Running 3 Alpha nodes, 3 Zero nodes, 1 Ratel node
Alpha nodes meshed together using linkerd (proxying gRPC)
Kubernetes 1.13.12 | node RAM: 8GB

### Steps to reproduce the issue (command/config used to run Dgraph).
Not able to reproduce: I exported the data from one of the alpha nodes. Using the `live loader` I imported the data in a fresh `stand-alone` cluster running the same version. I was **not** able to reproduce the issue this way. 

### Expected behaviour and actual result.

I expect the same result, but the second query returns no results (`[]`)
```
{
    works(func: type(Role)) @cascade  {
    	uid
    	user @filter(uid(0xf870)){
            uid
            id
        }
    }

    fails(func: type(Role)) @cascade  {
    	uid
    	user @filter(eq(id, "je4eke241")){
            uid
            id
        }
    }
}
```

**Actual result**
![image (4)](https://user-images.githubusercontent.com/2030968/82352560-74e30780-99fe-11ea-87d1-fdf197e7a016.png)

Schema
```
<id>:string @index(exact) .
<name>:string @index(exact) .
<type>:string @index(exact) .
<user>:[uid] @reverse .
<action>:string @index(exact) .
<parent>:uid @reverse .
<basedOn>:uid @reverse .
<resource>:[uid] @reverse .
<username>:string @index(exact) .
<references>:[uid] @reverse .
<dgraph.type>:[string] @index(exact) .
<permissions>:[uid] @reverse .
<organisation>:[uid] .
type Role {
	user
	resource
	permissions
	basedOn
}
type User {
	id
	name
}
type Resource {
	id
	type
	name
	parent
	references
}
type Permission {
	action
}
type RoleDefinition {
	name
	permissions
}

```

**Comment by anurags92** [comment]
Thanks @boedy I am taking a look, will get back when I have something.

**Comment by danielmai** [comment]
@boedy have you tried Dgraph v1.2.3?

**Comment by boedy** [comment]
This morning I did the following:

1. Upgraded the cluster to v1.2.4
2. Ran query: Got the expected response (**no bug**)
3. Noticed the release page stating we should not be using v1.2.4
4. Rolled back to v1.2.2
5. Ran query again: Got empty response (**bug still there**) 
6. Upgraded the cluster to v1.2.3
7. Ran query again: Got the expected response (**no bug**)

Given the fact that the problem persisted when I rolled back to v1.2.2 I'm assuming it's a bug related to that specific version or earlier versions.
